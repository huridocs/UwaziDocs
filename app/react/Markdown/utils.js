const objectPath = (path, object) =>
  path.split('.').reduce((o, key) => {
    if (!o || !key) {
      return o;
    }
    return o.toJS ? o.get(key) : o[key];
  }, object);

const logError = (err, propValueOf, propLabelOf) => {
  /* eslint-disable no-console */
  console.error('Error on EntityData: ');
  console.error('value-of: ', propValueOf, '; label-of: ', propLabelOf);
  console.error(err);
  /* eslint-enable no-console */
};

const validHtmlTags = new Set([
  'a',
  'abbr',
  'acronym',
  'address',
  'area',
  'article',
  'aside',
  'audio',
  'b',
  'bdi',
  'bdo',
  'big',
  'blink',
  'blockquote',
  'body',
  'br',
  'button',
  'canvas',
  'caption',
  'center',
  'circle',
  'cite',
  'clipPath',
  'code',
  'col',
  'colgroup',
  'content',
  'data',
  'datalist',
  'dd',
  'decorator',
  'defs',
  'del',
  'desc',
  'details',
  'dfn',
  'dir',
  'div',
  'dl',
  'dt',
  'element',
  'ellipse',
  'em',
  'embed',
  'feBlend',
  'feColorMatrix',
  'feComponentTransfer',
  'feComposite',
  'feConvolveMatrix',
  'feDiffuseLighting',
  'feDisplacementMap',
  'feDistantLight',
  'feFlood',
  'feFuncA',
  'feFuncB',
  'feFuncG',
  'feFuncR',
  'feGaussianBlur',
  'feImage',
  'feMerge',
  'feMergeNode',
  'feMorphology',
  'feOffset',
  'fePointLight',
  'feSpecularLighting',
  'feSpotLight',
  'feTile',
  'feTurbulence',
  'fieldset',
  'figcaption',
  'figure',
  'filter',
  'font',
  'footer',
  'foreignObject',
  'form',
  'g',
  'h1',
  'h2',
  'h3',
  'h4',
  'h5',
  'h6',
  'head',
  'header',
  'hgroup',
  'hr',
  'html',
  'i',
  'iframe',
  'image',
  'img',
  'input',
  'ins',
  'kbd',
  'label',
  'legend',
  'li',
  'line',
  'linearGradient',
  'main',
  'map',
  'mark',
  'marker',
  'marquee',
  'mask',
  'menu',
  'menuitem',
  'metadata',
  'meter',
  'nav',
  'nobr',
  'object',
  'ol',
  'optgroup',
  'option',
  'output',
  'p',
  'path',
  'pattern',
  'polygon',
  'polyline',
  'pre',
  'progress',
  'q',
  'radialGradient',
  'rect',
  'rp',
  'rt',
  'ruby',
  's',
  'samp',
  'section',
  'select',
  'shadow',
  'small',
  'source',
  'spacer',
  'span',
  'strike',
  'strong',
  'style',
  'sub',
  'summary',
  'sup',
  'svg',
  'switch',
  'symbol',
  'table',
  'tbody',
  'td',
  'template',
  'textarea',
  'tfoot',
  'th',
  'thead',
  'time',
  'tr',
  'track',
  'tspan',
  'tt',
  'u',
  'ul',
  'use',
  'var',
  'video',
  'view',
  'wbr',
]);

export { objectPath, logError, validHtmlTags };
