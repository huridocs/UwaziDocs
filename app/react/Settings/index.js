/* eslint-disable import/no-named-as-default */
import EntityTypesList from './components/EntityTypesList';
import FiltersForm from './components/FiltersForm';
import { RelationTypesList } from './components/RelationTypesList';
import ThesauriList from './components/ThesauriList';
import { Settings } from './Settings';
import SettingsAPI from './SettingsAPI';
import { Dashboard } from './components/Dashboard';
import { PreserveSettings } from './components/PreserveSettings';

export { CustomUploads } from './components/CustomUploads';

export {
  Settings,
  SettingsAPI,
  PreserveSettings,
  EntityTypesList,
  RelationTypesList,
  ThesauriList,
  FiltersForm,
  Dashboard,
};
