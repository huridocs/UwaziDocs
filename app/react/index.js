import React from 'react';
import ReactDOM from 'react-dom';

import * as Sentry from '@sentry/react';
import { BrowserTracing } from '@sentry/tracing';

import { AppContainer } from 'react-hot-loader';

import { SENTRY_APP_DSN } from 'app/config';
import App from './App';

import './App/sockets';

if (SENTRY_APP_DSN) {
  Sentry.init({
    dsn: SENTRY_APP_DSN,
    integrations: [new BrowserTracing()],
    tracesSampleRate: 0.1,
  });
}

const render = Component => {
  ReactDOM.hydrate(
    <AppContainer>
      <Component />
    </AppContainer>,
    document.getElementById('root')
  );
};

render(App);

if (module.hot) {
  module.hot.accept('./App', () => {
    const nextApp = require('./App');
    render(nextApp);
  });
}
