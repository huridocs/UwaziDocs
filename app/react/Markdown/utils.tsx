import { safeName } from 'shared/propertyNames';
import { showByType } from 'app/Metadata/components/Metadata';
import { ensure } from 'shared/tsUtils';
import { IImmutable } from 'shared/types/Immutable';
import { ClientTemplateSchema } from 'app/istore';
import { Translate } from 'app/I18N';
import React from 'react';
import { EntityDataProps } from './components/EntityData';

interface Options {
  formattedEntity: any;
  propertyName: string;
  newNameGeneration: boolean;
  template?: IImmutable<ClientTemplateSchema>;
}

const objectPath = (path: any, object: any) =>
  path.split('.').reduce((o: any, key: any) => {
    if (!o || !key) {
      return o;
    }
    return o.toJS ? o.get(key) : o[key];
  }, object);

const rootProperties = ['title', 'creationDate', 'editDate'];

const getPropertyData = ({ formattedEntity, propertyName, newNameGeneration }: Options) =>
  formattedEntity.metadata.find((p: any) => p.name === safeName(propertyName, newNameGeneration));

const extractRootProperty = ({ formattedEntity, propertyName }: Options) =>
  formattedEntity[propertyName];

const extractMetadataProperty = ({ formattedEntity, propertyName, newNameGeneration }: Options) => {
  const propertyData = formattedEntity.metadata.find(
    (p: any) => p.name === safeName(propertyName, newNameGeneration)
  );
  return showByType(propertyData, false);
};

const extractRootLabel = ({ propertyName, template: _template }: Options) => {
  const template = ensure<IImmutable<ClientTemplateSchema>>(_template);
  const term =
    template
      .get('commonProperties')
      ?.find(p => p?.get('name') === propertyName)
      .get('label') || '';

  const context = propertyName === 'title' ? template.get('_id') : 'System';
  return <Translate context={context}>{term}</Translate>;
};

const extractMetadataLabel = ({ formattedEntity, propertyName, newNameGeneration }: Options) => {
  const propertyData = getPropertyData({ formattedEntity, propertyName, newNameGeneration });
  return <Translate context={propertyData.translateContext}>{propertyData.label}</Translate>;
};

const logError = (err: any, propValueOf?: string, propLabelOf?: string) => {
  /* eslint-disable no-console */
  console.error('Error on EntityData: ');
  console.error('value-of: ', propValueOf, '; label-of: ', propLabelOf);
  console.error(err);
  /* eslint-enable no-console */
};

const getProperty = (
  propValueOf?: EntityDataProps['value-of'],
  propLabelOf?: EntityDataProps['label-of']
) => {
  if (propValueOf && propLabelOf) {
    throw new Error('Can\'t provide both "value-of" and "label-of".');
  }

  const property = propValueOf || propLabelOf;

  if (!property) {
    throw new Error('"value-of" or "label-of" must be provided.');
  }

  return property;
};

const getMethod = (propValueOf: string | undefined, propertyName: string) => {
  const isRootProperty = rootProperties.includes(propertyName);

  if (propValueOf) {
    return isRootProperty ? extractRootProperty : extractMetadataProperty;
  }

  return isRootProperty ? extractRootLabel : extractMetadataLabel;
};

export {
  objectPath,
  getPropertyData,
  extractRootProperty,
  extractMetadataProperty,
  extractRootLabel,
  extractMetadataLabel,
  logError,
  getProperty,
  getMethod,
};
