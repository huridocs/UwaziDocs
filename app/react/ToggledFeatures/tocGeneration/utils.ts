import { ClientSettings } from 'app/apiResponseTypes';
import { SearchParams } from 'shared/types/searchParams.d';

export const tocGenerationUtils = {
  aggregations(params: SearchParams, settings: ClientSettings) {
    return {
      ...params,
      ...(settings?.features?.tocGeneration ? { aggregateGeneratedToc: true } : {}),
    };
  },
};
