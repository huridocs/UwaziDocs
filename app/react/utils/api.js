import request from '../../shared/JSONRequest';
import { isClient } from 'app/utils';
import { APIURL } from '../config.js';
import { browserHistory } from 'react-router';
import { notify } from 'app/Notifications/actions/notificationsActions';
import { store } from 'app/store';
import loadingBar from 'app/App/LoadingProgressBar';

let cookie;
let locale;
let API_URL = APIURL;

const doneLoading = (data) => {
  loadingBar.done();
  return data;
};

const handleError = (url, method, error) => {
  if (!isClient) {
    return Promise.reject(error);
  }

  doneLoading();

  if (error.status === 401) {
    browserHistory.replace('/login');
    return Promise.reject(error);
  }

  if (error.status === 404) {
    browserHistory.replace('/404');
    return Promise.reject(error);
  }

  if (error.status === 500) {
    store.dispatch(notify('An error has occurred', 'danger'));
    return Promise.reject(error);
  }

  console.log(`API error without status code when calling to ${method} ${API_URL}${url}`);
  store.dispatch(notify(error.json.error, 'danger'));
  return Promise.reject(error);
};

export default {
  get: (url, data) => {
    loadingBar.start();
    return request.get(API_URL + url, data, { 'Content-language': locale, Cookie: cookie })
    .then(doneLoading)
    .catch(handleError.bind(null, url, 'GET'));
  },

  post: (url, data) => {
    loadingBar.start();
    return request.post(API_URL + url, data, { 'Content-language': locale })
    .then(doneLoading)
    .catch(handleError.bind(null, url, 'POST'));
  },

  delete: (url, data) => {
    loadingBar.start();
    return request.delete(API_URL + url, data, { 'Content-language': locale })
    .then(doneLoading)
    .catch(handleError.bind(null, url, 'DELETE'));
  },

  cookie(c) {
    cookie = c;
  },

  locale(key) {
    locale = key;
  },

  APIURL(url) {
    API_URL = url;
  }
};
