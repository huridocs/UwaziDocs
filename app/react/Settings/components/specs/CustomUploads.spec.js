/**
 * @jest-environment jsdom
 */
import React from 'react';
import Immutable from 'immutable';
import { screen, fireEvent } from '@testing-library/react';

import api from 'app/utils/api';
import { RequestParams } from 'app/utils/RequestParams';
import { renderConnectedContainer, defaultState } from 'app/utils/test/renderConnected';
import { CustomUploadsComponent as CustomUploads, mapStateToProps } from '../CustomUploads';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => jest.fn(),
}));

describe('CustomUploads', () => {
  let props;
  let renderResult;

  beforeEach(() => {
    spyOn(api, 'get').and.callFake(async () => Promise.resolve({ json: 'uploads' }));
    props = {
      customUploads: Immutable.fromJS([]),
      deleteCustomUpload: jasmine.createSpy('deleteCustomUpload'),
      upload: jasmine.createSpy('upload'),
      params: { lang: 'es' },
    };
  });

  const render = () => {
    const reduxStore = {
      ...defaultState,
    };

    ({ renderResult } = renderConnectedContainer(
      <CustomUploads {...props} />,
      () => reduxStore,
      'MemoryRouter',
      [{ pathname: '/test' }]
    ));
  };

  it('should render CustomUploads component with uploaded files', () => {
    props.customUploads = Immutable.fromJS([{ filename: 'file1' }, { filename: 'file2' }]);
    render();
    expect(renderResult).toMatchSnapshot();
  });

  describe('when upload on progress', () => {
    it('should render on progress feedback', () => {
      props.progress = true;
      render();
      expect(renderResult).toMatchSnapshot();
    });
  });

  describe('deleteCustomUpload', () => {
    it('should call deleteCustomUpload on click', () => {
      props.customUploads = Immutable.fromJS([{ _id: 'upload', filename: 'name' }]);
      render();
      fireEvent.click(screen.getByRole('button', { name: 'Delete' }));
      fireEvent.click(screen.getByRole('button', { name: 'Accept' }));
      expect(props.deleteCustomUpload).toHaveBeenCalledWith('upload');
    });
  });

  describe('mapStateToProps', () => {
    it('should map current progress and files to props', () => {
      const state = {
        customUploads: 'customUploads',
        progress: Immutable.fromJS({}),
      };

      props = mapStateToProps(state);
      expect(props.customUploads).toBe('customUploads');
      expect(props.progress).toBe(false);

      state.progress = Immutable.fromJS({
        customUpload_unique_id: 1,
        customUpload_unique_id2: 100,
      });
      props = mapStateToProps(state);
      expect(props.progress).toBe(true);

      state.progress = Immutable.fromJS({ not_custom_upload: 9 });
      props = mapStateToProps(state);
      expect(props.progress).toBe(false);
    });
  });

  describe('requestState', () => {
    it('should get the uploads', async () => {
      const request = new RequestParams();
      const actions = await CustomUploads.requestState(request);

      const expectedParams = new RequestParams({ type: 'custom' });
      expect(api.get).toHaveBeenCalledWith('files', expectedParams);
      expect(actions).toMatchSnapshot();
    });
  });
});
