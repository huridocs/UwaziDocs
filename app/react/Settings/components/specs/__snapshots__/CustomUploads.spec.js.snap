// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomUploads requestState should get the uploads 1`] = `
Array [
  Object {
    "type": "customUploads/SET",
    "value": "uploads",
  },
]
`;

exports[`CustomUploads should render CustomUploads component with uploaded files 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <CustomUploads
    customUploads={
      Immutable.List [
        Immutable.Map {
          "filename": "file1",
        },
        Immutable.Map {
          "filename": "file2",
        },
      ]
    }
    deleteCustomUpload={[Function]}
    params={
      Object {
        "lang": "es",
      }
    }
    progress={false}
    upload={[Function]}
  >
    <div
      className="settings-content without-footer"
    >
      <div
        className="panel panel-default"
      >
        <SettingsHeader>
          <div
            className="panel-heading"
          >
            <Connect(I18NLink)
              className="only-mobile"
              to="settings/"
            >
              <I18NLink
                className="only-mobile"
                confirmMessage=""
                confirmTitle=""
                disabled={false}
                dispatch={[Function]}
                onClick={[Function]}
                to="/en/settings/"
              >
                <Link
                  className="only-mobile"
                  disabled={false}
                  onClick={[Function]}
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/en/settings/"
                >
                  <a
                    className="only-mobile"
                    disabled={false}
                    onClick={[Function]}
                    style={Object {}}
                  >
                    <Connect(Icon)
                      directionAware={true}
                      icon="arrow-left"
                    >
                      <Icon
                        directionAware={true}
                        icon="arrow-left"
                        locale="en"
                      >
                        <FontAwesomeIcon
                          beat={false}
                          beatFade={false}
                          border={false}
                          bounce={false}
                          className=""
                          fade={false}
                          fixedWidth={false}
                          flip={null}
                          icon="arrow-left"
                          inverse={false}
                          listItem={false}
                          mask={null}
                          maskId={null}
                          pull={null}
                          pulse={false}
                          rotation={null}
                          shake={false}
                          size={null}
                          spin={false}
                          spinPulse={false}
                          spinReverse={false}
                          swapOpacity={false}
                          symbol={false}
                          title=""
                          titleId={null}
                          transform={null}
                        >
                          <svg
                            aria-hidden="true"
                            className="svg-inline--fa fa-arrow-left fa-w-14 "
                            data-icon="arrow-left"
                            data-prefix="fas"
                            focusable="false"
                            role="img"
                            style={Object {}}
                            viewBox="0 0 448 512"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
                              fill="currentColor"
                              style={Object {}}
                            />
                          </svg>
                        </FontAwesomeIcon>
                      </Icon>
                    </Connect(Icon)>
                    <span
                      className="btn-label"
                    >
                      <Connect(Translate)>
                        <Translate
                          context="System"
                          edit={[Function]}
                          i18nmode={true}
                          translationKey="Back"
                        >
                          <span
                            className="translation active"
                            onClick={[Function]}
                          >
                            Back
                          </span>
                        </Translate>
                      </Connect(Translate)>
                    </span>
                  </a>
                </Link>
              </I18NLink>
            </Connect(I18NLink)>
            <Connect(Translate)>
              <Translate
                context="System"
                edit={[Function]}
                i18nmode={true}
                translationKey="Custom Uploads"
              >
                <span
                  className="translation active"
                  onClick={[Function]}
                >
                  Custom Uploads
                </span>
              </Translate>
            </Connect(Translate)>
          </div>
        </SettingsHeader>
        <div
          className="panel-body custom-uploads"
        >
          <Dropzone
            autoFocus={false}
            className="upload-box"
            disabled={false}
            getFilesFromEvent={[Function]}
            maxFiles={0}
            maxSize={Infinity}
            minSize={0}
            multiple={true}
            noClick={false}
            noDrag={false}
            noDragEventsBubbling={false}
            noKeyboard={false}
            onDrop={[Function]}
            preventDropOnDocument={true}
            useFsAccessApi={true}
            validator={null}
          >
            <div
              onBlur={[Function]}
              onClick={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="presentation"
              tabIndex={0}
            >
              <div
                className="upload-box_wrapper"
              >
                <Connect(Icon)
                  icon="upload"
                >
                  <Icon
                    directionAware={false}
                    icon="upload"
                    locale="en"
                  >
                    <FontAwesomeIcon
                      beat={false}
                      beatFade={false}
                      border={false}
                      bounce={false}
                      className=""
                      fade={false}
                      fixedWidth={false}
                      flip={null}
                      icon="upload"
                      inverse={false}
                      listItem={false}
                      mask={null}
                      maskId={null}
                      pull={null}
                      pulse={false}
                      rotation={null}
                      shake={false}
                      size={null}
                      spin={false}
                      spinPulse={false}
                      spinReverse={false}
                      swapOpacity={false}
                      symbol={false}
                      title=""
                      titleId={null}
                      transform={null}
                    >
                      <svg
                        aria-hidden="true"
                        className="svg-inline--fa fa-upload fa-w-16 "
                        data-icon="upload"
                        data-prefix="fas"
                        focusable="false"
                        role="img"
                        style={Object {}}
                        viewBox="0 0 512 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M296 384h-80c-13.3 0-24-10.7-24-24V192h-87.7c-17.8 0-26.7-21.5-14.1-34.1L242.3 5.7c7.5-7.5 19.8-7.5 27.3 0l152.2 152.2c12.6 12.6 3.7 34.1-14.1 34.1H320v168c0 13.3-10.7 24-24 24zm216-8v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h136v8c0 30.9 25.1 56 56 56h80c30.9 0 56-25.1 56-56v-8h136c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
                          fill="currentColor"
                          style={Object {}}
                        />
                      </svg>
                    </FontAwesomeIcon>
                  </Icon>
                </Connect(Icon)>
                <input
                  multiple={true}
                  onChange={[Function]}
                  onClick={[Function]}
                  style={
                    Object {
                      "display": "none",
                    }
                  }
                  tabIndex={-1}
                  type="file"
                />
                <button
                  className="upload-box_link"
                  type="button"
                >
                  <Connect(Translate)>
                    <Translate
                      context="System"
                      edit={[Function]}
                      i18nmode={true}
                      translationKey="Browse files to upload"
                    >
                      <span
                        className="translation active"
                        onClick={[Function]}
                      >
                        Browse files to upload
                      </span>
                    </Translate>
                  </Connect(Translate)>
                </button>
                <span>
                   
                  <Connect(Translate)>
                    <Translate
                      context="System"
                      edit={[Function]}
                      i18nmode={true}
                      translationKey="or drop your files here."
                    >
                      <span
                        className="translation active"
                        onClick={[Function]}
                      >
                        or drop your files here.
                      </span>
                    </Translate>
                  </Connect(Translate)>
                </span>
              </div>
            </div>
          </Dropzone>
          <ul>
            <li
              key="file1"
            >
              <Thumbnail
                alt=""
                file="/assets/file1"
              >
                <div
                  className="thumbnail"
                />
              </Thumbnail>
              <div
                className="info"
              >
                <span
                  no-translate="true"
                >
                  URL:
                </span>
                <span
                  className="thumbnail-url"
                >
                  /assets/file1
                </span>
                <ConfirmButton
                  action={[Function]}
                  className=""
                  message="Are you sure you want to continue?"
                  title="Confirm action"
                >
                  <button
                    className=""
                    onClick={[Function]}
                    type="button"
                  >
                    <Connect(Translate)>
                      <Translate
                        context="System"
                        edit={[Function]}
                        i18nmode={true}
                        translationKey="Delete"
                      >
                        <span
                          className="translation active"
                          onClick={[Function]}
                        >
                          Delete
                        </span>
                      </Translate>
                    </Connect(Translate)>
                  </button>
                </ConfirmButton>
              </div>
            </li>
            <li
              key="file2"
            >
              <Thumbnail
                alt=""
                file="/assets/file2"
              >
                <div
                  className="thumbnail"
                />
              </Thumbnail>
              <div
                className="info"
              >
                <span
                  no-translate="true"
                >
                  URL:
                </span>
                <span
                  className="thumbnail-url"
                >
                  /assets/file2
                </span>
                <ConfirmButton
                  action={[Function]}
                  className=""
                  message="Are you sure you want to continue?"
                  title="Confirm action"
                >
                  <button
                    className=""
                    onClick={[Function]}
                    type="button"
                  >
                    <Connect(Translate)>
                      <Translate
                        context="System"
                        edit={[Function]}
                        i18nmode={true}
                        translationKey="Delete"
                      >
                        <span
                          className="translation active"
                          onClick={[Function]}
                        >
                          Delete
                        </span>
                      </Translate>
                    </Connect(Translate)>
                  </button>
                </ConfirmButton>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </CustomUploads>
</Provider>
`;

exports[`CustomUploads when upload on progress should render on progress feedback 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <CustomUploads
    customUploads={Immutable.List []}
    deleteCustomUpload={[Function]}
    params={
      Object {
        "lang": "es",
      }
    }
    progress={true}
    upload={[Function]}
  >
    <div
      className="settings-content without-footer"
    >
      <div
        className="panel panel-default"
      >
        <SettingsHeader>
          <div
            className="panel-heading"
          >
            <Connect(I18NLink)
              className="only-mobile"
              to="settings/"
            >
              <I18NLink
                className="only-mobile"
                confirmMessage=""
                confirmTitle=""
                disabled={false}
                dispatch={[Function]}
                onClick={[Function]}
                to="/en/settings/"
              >
                <Link
                  className="only-mobile"
                  disabled={false}
                  onClick={[Function]}
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/en/settings/"
                >
                  <a
                    className="only-mobile"
                    disabled={false}
                    onClick={[Function]}
                    style={Object {}}
                  >
                    <Connect(Icon)
                      directionAware={true}
                      icon="arrow-left"
                    >
                      <Icon
                        directionAware={true}
                        icon="arrow-left"
                        locale="en"
                      >
                        <FontAwesomeIcon
                          beat={false}
                          beatFade={false}
                          border={false}
                          bounce={false}
                          className=""
                          fade={false}
                          fixedWidth={false}
                          flip={null}
                          icon="arrow-left"
                          inverse={false}
                          listItem={false}
                          mask={null}
                          maskId={null}
                          pull={null}
                          pulse={false}
                          rotation={null}
                          shake={false}
                          size={null}
                          spin={false}
                          spinPulse={false}
                          spinReverse={false}
                          swapOpacity={false}
                          symbol={false}
                          title=""
                          titleId={null}
                          transform={null}
                        >
                          <svg
                            aria-hidden="true"
                            className="svg-inline--fa fa-arrow-left fa-w-14 "
                            data-icon="arrow-left"
                            data-prefix="fas"
                            focusable="false"
                            role="img"
                            style={Object {}}
                            viewBox="0 0 448 512"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
                              fill="currentColor"
                              style={Object {}}
                            />
                          </svg>
                        </FontAwesomeIcon>
                      </Icon>
                    </Connect(Icon)>
                    <span
                      className="btn-label"
                    >
                      <Connect(Translate)>
                        <Translate
                          context="System"
                          edit={[Function]}
                          i18nmode={true}
                          translationKey="Back"
                        >
                          <span
                            className="translation active"
                            onClick={[Function]}
                          >
                            Back
                          </span>
                        </Translate>
                      </Connect(Translate)>
                    </span>
                  </a>
                </Link>
              </I18NLink>
            </Connect(I18NLink)>
            <Connect(Translate)>
              <Translate
                context="System"
                edit={[Function]}
                i18nmode={true}
                translationKey="Custom Uploads"
              >
                <span
                  className="translation active"
                  onClick={[Function]}
                >
                  Custom Uploads
                </span>
              </Translate>
            </Connect(Translate)>
          </div>
        </SettingsHeader>
        <div
          className="panel-body custom-uploads"
        >
          <Dropzone
            autoFocus={false}
            className="upload-box"
            disabled={false}
            getFilesFromEvent={[Function]}
            maxFiles={0}
            maxSize={Infinity}
            minSize={0}
            multiple={true}
            noClick={false}
            noDrag={false}
            noDragEventsBubbling={false}
            noKeyboard={false}
            onDrop={[Function]}
            preventDropOnDocument={true}
            useFsAccessApi={true}
            validator={null}
          >
            <div
              onBlur={[Function]}
              onClick={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="presentation"
              tabIndex={0}
            >
              <div
                className="upload-box_wrapper"
              >
                <Connect(Icon)
                  icon="upload"
                >
                  <Icon
                    directionAware={false}
                    icon="upload"
                    locale="en"
                  >
                    <FontAwesomeIcon
                      beat={false}
                      beatFade={false}
                      border={false}
                      bounce={false}
                      className=""
                      fade={false}
                      fixedWidth={false}
                      flip={null}
                      icon="upload"
                      inverse={false}
                      listItem={false}
                      mask={null}
                      maskId={null}
                      pull={null}
                      pulse={false}
                      rotation={null}
                      shake={false}
                      size={null}
                      spin={false}
                      spinPulse={false}
                      spinReverse={false}
                      swapOpacity={false}
                      symbol={false}
                      title=""
                      titleId={null}
                      transform={null}
                    >
                      <svg
                        aria-hidden="true"
                        className="svg-inline--fa fa-upload fa-w-16 "
                        data-icon="upload"
                        data-prefix="fas"
                        focusable="false"
                        role="img"
                        style={Object {}}
                        viewBox="0 0 512 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M296 384h-80c-13.3 0-24-10.7-24-24V192h-87.7c-17.8 0-26.7-21.5-14.1-34.1L242.3 5.7c7.5-7.5 19.8-7.5 27.3 0l152.2 152.2c12.6 12.6 3.7 34.1-14.1 34.1H320v168c0 13.3-10.7 24-24 24zm216-8v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h136v8c0 30.9 25.1 56 56 56h80c30.9 0 56-25.1 56-56v-8h136c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
                          fill="currentColor"
                          style={Object {}}
                        />
                      </svg>
                    </FontAwesomeIcon>
                  </Icon>
                </Connect(Icon)>
                <input
                  multiple={true}
                  onChange={[Function]}
                  onClick={[Function]}
                  style={
                    Object {
                      "display": "none",
                    }
                  }
                  tabIndex={-1}
                  type="file"
                />
                <button
                  className="upload-box_link"
                  type="button"
                >
                  <Connect(Translate)>
                    <Translate
                      context="System"
                      edit={[Function]}
                      i18nmode={true}
                      translationKey="Browse files to upload"
                    >
                      <span
                        className="translation active"
                        onClick={[Function]}
                      >
                        Browse files to upload
                      </span>
                    </Translate>
                  </Connect(Translate)>
                </button>
                <span>
                   
                  <Connect(Translate)>
                    <Translate
                      context="System"
                      edit={[Function]}
                      i18nmode={true}
                      translationKey="or drop your files here."
                    >
                      <span
                        className="translation active"
                        onClick={[Function]}
                      >
                        or drop your files here.
                      </span>
                    </Translate>
                  </Connect(Translate)>
                </span>
              </div>
              <div
                className="uploading"
              >
                <Connect(Icon)
                  icon="spinner"
                  spin={true}
                >
                  <Icon
                    directionAware={false}
                    icon="spinner"
                    locale="en"
                    spin={true}
                  >
                    <FontAwesomeIcon
                      beat={false}
                      beatFade={false}
                      border={false}
                      bounce={false}
                      className=""
                      fade={false}
                      fixedWidth={false}
                      flip={null}
                      icon="spinner"
                      inverse={false}
                      listItem={false}
                      mask={null}
                      maskId={null}
                      pull={null}
                      pulse={false}
                      rotation={null}
                      shake={false}
                      size={null}
                      spin={true}
                      spinPulse={false}
                      spinReverse={false}
                      swapOpacity={false}
                      symbol={false}
                      title=""
                      titleId={null}
                      transform={null}
                    >
                      <svg
                        aria-hidden="true"
                        className="svg-inline--fa fa-spinner fa-w-16 fa-spin "
                        data-icon="spinner"
                        data-prefix="fas"
                        focusable="false"
                        role="img"
                        style={Object {}}
                        viewBox="0 0 512 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
                          fill="currentColor"
                          style={Object {}}
                        />
                      </svg>
                    </FontAwesomeIcon>
                  </Icon>
                </Connect(Icon)>
                <Connect(Translate)>
                  <Translate
                    context="System"
                    edit={[Function]}
                    i18nmode={true}
                    translationKey="Uploading ..."
                  >
                    <span
                      className="translation active"
                      onClick={[Function]}
                    >
                      Uploading ...
                    </span>
                  </Translate>
                </Connect(Translate)>
              </div>
            </div>
          </Dropzone>
          <ul>
            <Component />
          </ul>
        </div>
      </div>
    </div>
  </CustomUploads>
</Provider>
`;
